endpoints/openrtb2/video_auction.go

```diff
@@ -0,0 +1,299 @@
+	if err != nil {
+		return
+	}
+
+	//load additional data - stored simplified req
+	storedRequestId, err := getVideoStoredRequestId(requestJson)
+	if err != nil {
+		return
+	}
+	storedRequest, err := loadStoredVideoRequest(storedRequestId)
+	if err != nil {
+		return
+	}
+
+	//merge incoming req with stored video req
+	resolvedRequest, err := jsonpatch.MergePatch(requestJson, storedRequest)
+
+	//unmarshal and validate combined result
+	videoBidReq, errl := deps.parseVideoRequest(resolvedRequest)
+	if len(errl) > 0 {
+		return
+	}
+
+	var bidReq = openrtb.BidRequest{}
+
+	// Populate any "missing" OpenRTB fields with info from other sources, (e.g. HTTP request headers).
+	deps.setFieldsImplicitly(r, &bidReq)
+
+	//create full open rtb req from full video request
+	mergeData(&videoBidReq, &bidReq)
+
+	//create impressions array
+	imps, errl := createImpressions(videoBidReq)
+	bidReq.Imp = imps
+
+	errL := deps.validateRequest(&bidReq)
+	if len(errL) > 0 {
+		//handle errors
+		return
+	}
+
+	labels := pbsmetrics.Labels{}
+	//execute auction logic
+	response, err := deps.ex.HoldAuction(nil, &bidReq, nil, labels, &deps.categories)
+	if err != nil {
+		//handle error
+	}
+
+	//build simplified response
+	var bidResp = openrtb_ext.BidResponseVideo{}
+	buildVideoResponse(response, &bidResp)
+
+}
+
+func createImpressions(videoReq openrtb_ext.BidRequestVideo) (imps []openrtb.Imp, errs []error) {
+	videoDur := videoReq.PodConfig.DurationRangeSec
+	minDuration, maxDuration := minMax(videoDur)
+	reqExactDur := videoReq.PodConfig.RequireExactDuration
+	videoData := videoReq.Video
+
+	finalImpsArray := make([]openrtb.Imp, 0)
+	for podIndex, pod := range videoReq.PodConfig.Pods {
+
+		//load stored impression
+		storedImpressionId := string(pod.ConfigId)
+		storedImp := loadStoredImp(storedImpressionId)
+
+		impTemplate := createImpressionTemplate(storedImp, videoData)
+		numImps := pod.AdPodDurationSec / minDuration
+
+		impDivNumber := numImps / len(videoDur)
+
+		impsArray := make([]openrtb.Imp, numImps)
+		for impInd := range impsArray {
+			impsArray[impInd] = impTemplate
+			if reqExactDur {
+				//floor := int(math.Floor(ind/impDivNumber))
+				floor := impInd / impDivNumber
+				impsArray[impInd].Video.MaxDuration = int64(videoDur[floor])
+				impsArray[impInd].Video.MinDuration = int64(videoDur[floor])
+				fmt.Println(podIndex, "  ", impInd, "duration ", videoDur[floor])
+			} else {
+				impsArray[impInd].Video.MaxDuration = int64(maxDuration)
+			}
+
+			impsArray[impInd].ID = fmt.Sprintf("pod%d_%d", podIndex, impInd)
+		}
+		finalImpsArray = append(finalImpsArray, impsArray...)
+
+	}
+	return finalImpsArray, nil
+}
+
+func createImpressionTemplate(imp openrtb.Imp, video openrtb_ext.SimplifiedVideo) openrtb.Imp {
+	imp.Video = &openrtb.Video{
+		nil,
+		0,
+		0,
+		nil,
+		0,
+		0,
+		0,
+		nil,
+		0,
+		0,
+		nil,
+		0,
+		0,
+		0,
+		nil,
+		0,
+		0,
+		0,
+		0,
+		nil,
+		0,
+		nil,
+		nil,
+		nil,
+		nil,
+		nil,
+		nil,
+	}
```

Comment:
You should be able to omit those parameters, undeclared fields get assigned their zero value by default in Go.

Reply:
Done